helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -86,13 +86,22 @@ processors:
   # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor
   resourcedetection:
     detectors:
+      # Note: Kubernetes detectors need to come first so they set the proper cloud.platform
+      # before it gets set later be gce/ec2.
       - env
-      {{- if eq .Values.platform "gcp" }}
+      {{- if eq (.Values.distro | default "") "gke" }}
       - gke
-      - gce
-      {{- else if eq .Values.platform "aws" }}
+      {{- else if eq (.Values.distro | default "") "eks" }}
       - eks
+      {{- else if eq (.Values.distro | default "") "aks" }}
+      - aks
+      {{- end }}
+      {{- if eq (.Values.provider | default "") "gcp" }}
+      - gce
+      {{- else if eq (.Values.provider | default "") "aws" }}
```

Comment:
can we set `""` as the default value in values.yaml instead of `| default ""`?

Reply:
Yeah I was thinking about case where somebody did

```yaml
provider:
```

Which would set it to null but now I realize having it in the schema as required and non-nullable would prevent that. Seems to work.